src/main/java/com/rybka/view/ExchangeView.java

```diff
@@ -1,40 +1,57 @@
 package com.rybka.view;
 
+import com.rybka.dao.HibernateDAO;
+import com.rybka.model.Currency;
 import com.rybka.service.ExchangeService;
+import lombok.extern.log4j.Log4j;
 
 import java.util.Scanner;
 
+@Log4j
 public class ExchangeView {
 
     private final Scanner scanner = new Scanner(System.in);
     private final ExchangeService service = new ExchangeService();
+    private final HibernateDAO hibernateDAO = new HibernateDAO();
 
     public void showDialog() {
 
         try {
             System.out.print("Insert base currency: ");
-            String userBaseCurrencyAbbreviation = scanner.next().toUpperCase();
+            var userBaseCurrency = scanner.next().toUpperCase();
             System.out.print("Enter value: ");
-            Double userValue = scanner.nextDouble();
+            var userValue = scanner.nextDouble();
             System.out.print("Insert target currency: ");
 
-            String userTargetCurrencyAbbreviation = scanner.next().toUpperCase();
+            var userTargetCurrency = scanner.next().toUpperCase();
 
-            service.loadCurrencyOf(userBaseCurrencyAbbreviation, userTargetCurrencyAbbreviation);
-            service.exchange(userValue);
-            service.getCurrencyObject();
+            var currencyResponse = service.loadCurrencyOf(userBaseCurrency, userTargetCurrency);
+            log.info(currencyResponse);
 
-            getResult();
+            var total = service.calculateTotal(currencyResponse.getRate(), userValue);
+            var convertedResult = constructConvertedResult(currencyResponse.getBase(),
+                    currencyResponse.getTarget(),
+                    userValue,
+                    currencyResponse.getRate(),
+                    total);
+
+            hibernateDAO.save(convertedResult);
+            hibernateDAO.showTableRow();
+
+            showExchange(convertedResult);
         } catch (Exception e) {
-            System.out.println("Incorrect input!");
-            System.exit(2);
+            log.error("Some issues are occurred! Reason: " + e.getMessage());
         }
     }
 
-    public void getResult() {
-        System.out.println(String.format("%.4f %s -> %.4f %s", service.getAmount(),
-                service.getBaseCurrencyAbbreviation(), service.getTotal(),
-                service.getTargetCurrencyAbbreviation()));
+    private void showExchange(Currency currency) {
+        log.info(String.format("%.4f %s -> %.4f %s", currency.getAmount(),
+                currency.getBase(), currency.getTotal(),
+                currency.getTarget()));
+    }
+
+    private Currency constructConvertedResult(String base, String target, double amount, double rate, double total) {
+        return new Currency(base, target, amount, rate, total);
```

Comment:
As we discussed this should we should save exchange history. So the object name should be `CurrencyHistory`. Don't forget to add additional field `date`.

Reply:
fixed. There is date field as timestamp